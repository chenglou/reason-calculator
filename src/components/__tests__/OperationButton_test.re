open Jest;
open Expect;

let _ =

describe "OperationButton" (fun _ => {
  test "renders" (fun _ => {
    let action = Action.Add;
    let dispatch = fun _ => ();
    let tree = OperationButton.createElement ::action ::dispatch children::[] ()
      |> ReactShallowRenderer.renderWithRenderer;

    expect tree |> toMatchSnapshot;
  });

  test "renders a class name" (fun _ => {
    let action = Action.Add;
    let className = Styles.make color::"rebeccapurple" () |> Styles.className;
    let dispatch = fun _ => ();
    let tree = OperationButton.createElement ::action ::className ::dispatch children::[] ()
      |> ReactShallowRenderer.renderWithRenderer;

    expect tree |> toMatchSnapshot;
  });
});
