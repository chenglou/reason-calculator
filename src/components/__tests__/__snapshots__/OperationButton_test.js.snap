// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperationButton renders 1`] = `
<button
  className="css-1uxa59e css-1efi2jj css-1c4ilrr css-1wf9u1s css-wlhq18"
  dangerouslySetInnerHTML={
    Object {
      "__html": "+",
    }
  }
  onClick={[Function]}
/>
`;

exports[`OperationButton renders a class name 1`] = `
<button
  className="css-1uxa59e css-1efi2jj css-1c4ilrr css-aori8y"
  dangerouslySetInnerHTML={
    Object {
      "__html": "+",
    }
  }
  onClick={[Function]}
/>
`;
